isabelle/Examples/Escrow.thy

```diff
@@ -0,0 +1,252 @@
+(*<*)
+theory Escrow
+  imports Core.Semantics Core.TransactionBound Core.Timeout
+begin
+(*>*)
+section \<open>Escrow contract\<close>
+text \<open>An escrow contract is for the purchase of an item at a price.
+The buyer may complain that there is a problem, asking for a refund.
+If the seller disputes that complaint, then a mediator decides.
+\<close>
+subsection \<open>Contract definition\<close>
+text \<open>In the escrow contract there are the following roles:\<close>
+definition "seller = Role (BS ''Seller'')"
+definition "buyer = Role (BS ''Buyer'')"
+definition "mediator = Role (BS ''Mediator'')"
+text \<open>An escrow contract takes as arguments the price, token
+and the deadlines that apply for the payment, the complaint, the dispute and
+the mediation.
+\<close>
```

Comment:
We are inside the `Contract escrow` section, we could reduce the number of times we say it.

```suggestion
text \<open>The participants of the contract are:\<close>
definition "seller = Role (BS ''Seller'')"
definition "buyer = Role (BS ''Buyer'')"
definition "mediator = Role (BS ''Mediator'')"
text \<open>An escrow contract takes as arguments the price, token
and the deadlines that apply for the payment, the complaint, the dispute and
the mediation.
\<close>
```